### PR TITLE
Install latest patch versions of golang for windows

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
- install latest golang version for windows
#63 
#458

### CODE REVIEW

File: images/win/Windows2019-Azure.json [hunk 1]
```diff
@@ -282,8 +282,8 @@
         {
             "type": "powershell",
             "environment_vars": [
-                "GO_VERSIONS=1.10.8,1.11.12,1.12.7,1.13",
-                "GO_DEFAULT=1.12.7"
+                "GO_VERSIONS=1.10,1.11,1.13",
```
Comment (al-cheb) [inline images/win/Windows2019-Azure.json]: Missing 1.12

File: images/win/Windows2019-Azure.json [hunk 2]
```diff
@@ -591,8 +591,8 @@
         {
             "type": "powershell",
             "environment_vars": [
-                "GO_VERSIONS=1.10.8,1.11.12,1.12.7,1.13",
-                "GO_DEFAULT=1.12.7"
+                "GO_VERSIONS=1.10,1.11,1.12,1.13",
+                "GO_DEFAULT=1.12"
```
Comment (al-cheb) [inline images/win/Windows2019-Azure.json]: Incorrect default version

File: images/win/Windows2019-Azure.json (hunk 1 above)
Comment (dmitry-shibanov) [reply images/win/Windows2019-Azure.json]: Done.

File: images/win/Windows2019-Azure.json (hunk 2 above)
Comment (dmitry-shibanov) [reply images/win/Windows2019-Azure.json]: Done.

File: images/win/scripts/Installers/Install-Go.ps1 [hunk 3]
```diff
@@ -13,17 +13,20 @@ function Install-GoVersion
         [Switch]$addToDefaultPath
     )
 
+    $latestVersionObject = $refsJson | Where-Object { $_.ref -Match "refs/tags/go$goVersion[./d]*" }  | Select-Object -Last 1
+    $latestVersion = $latestVersionObject.ref -replace "\D+\D+[./d]*"
```
Comment (al-cheb) [inline images/win/scripts/Installers/Install-Go.ps1]: nitpick: $latestVersionObject.ref.replace('refs/tags/go','') is simple than regex

File: images/win/scripts/Installers/Validate-Go.ps1 [hunk 4]
```diff
@@ -8,18 +8,25 @@ function Get-GoVersion
 {
     Param
     (
-        [String]$goRootPath
+        [String]$goVersion
     )
+    $goDirectory = Get-ChildItem -Path $env:SystemDrive -Force -Filter "Go$goVersion*" | Select-Object -First 1
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Validate-Go.ps1]: Why we need -Force flag for Get-ChildItem?

File: images/win/scripts/Installers/Validate-Go.ps1 [hunk 5]
```diff
@@ -8,18 +8,25 @@ function Get-GoVersion
...
 
-    $env:Path = "$goRootPath\bin;" + $env:Path
-    if( $(go version) -match 'go version go(?<version>.*) win.*' )
+    $env:Path = "$goPath\bin;" + $env:Path
+    $version = $(go version)
+
+    $semanticEquality = $version -match 'go version go(?<version>.*) win.*'
+    $matchVersion = $version -match $goVersion
+
+    if( $semanticEquality -And $matchVersion)
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Validate-Go.ps1]: nitpick: reduntant space

File: images/win/scripts/Installers/Install-Go.ps1 [hunk 6]
```diff
@@ -34,25 +37,25 @@ function Install-GoVersion
...
     # Rename the extracted "go" directory to include the Go version number (to support side-by-side versions of Go).
-    $newDirName = "Go$goVersion"
+    $newDirName = "Go$latestVersion"
     Rename-Item -path "C:\go" -newName $newDirName
 
     # Delete the Go zip archive.
-    Write-Host "Deleting downloaded archive of Go $goVersion..."
-    Remove-Item go$goVersion.windows-a`md64.zip
+    Write-Host "Deleting downloaded archive of Go $latestVersion..."
+    Remove-Item go$latestVersion.windows-a`md64.zip
```
Comment (al-cheb) [inline images/win/scripts/Installers/Install-Go.ps1]: Why do we need ` ?

File: images/win/scripts/Installers/Install-Go.ps1 [hunk 7]
```diff
@@ -4,7 +4,7 @@
 ################################################################################
 
 Import-Module -Name ImageHelpers -Force
-
+$refsJson = Invoke-RestMethod "https://api.github.com/repos/golang/go/git/refs/tags"
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Install-Go.ps1]: nitpick: add space before function

File: images/win/Windows2019-Azure.json [hunk 8]
```diff
@@ -591,8 +591,8 @@
         {
             "type": "powershell",
             "environment_vars": [
-                "GO_VERSIONS=1.10.8,1.11.12,1.12.7,1.13",
-                "GO_DEFAULT=1.12.7"
+                "GO_VERSIONS=1.10,1.11,1.12,1.13",
```
Comment (maxim-lobanov) [inline images/win/Windows2019-Azure.json]: Could you please add spaces between versions for readability?
`"GO_VERSIONS=1.10, 1.11, 1.12, 1.13",`
We can trim lines later

File: images/win/scripts/Installers/Install-Go.ps1 (hunk 3 above)
Comment (dmitry-shibanov) [reply images/win/scripts/Installers/Install-Go.ps1]: Done.

File: images/win/scripts/Installers/Validate-Go.ps1 (hunk 5 above)
Comment (dmitry-shibanov) [reply images/win/scripts/Installers/Validate-Go.ps1]: Done.

File: images/win/scripts/Installers/Validate-Go.ps1 (hunk 4 above)
Comment (dmitry-shibanov) [reply images/win/scripts/Installers/Validate-Go.ps1]: Done.

File: images/win/scripts/Installers/Install-Go.ps1 (hunk 6 above)
Comment (dmitry-shibanov) [reply images/win/scripts/Installers/Install-Go.ps1]: Done.

File: images/win/Windows2019-Azure.json (hunk 8 above)
Comment (dmitry-shibanov) [reply images/win/Windows2019-Azure.json]: Done.

File: images/win/scripts/Installers/Install-Go.ps1 (hunk 7 above)
Comment (dmitry-shibanov) [reply images/win/scripts/Installers/Install-Go.ps1]: Done.

**Pull request merged**